src/python/pants/backend/docker/goals/package_image_test.py

```diff
@@ -646,3 +651,103 @@ def test_docker_build_fail_logs(
     assert_logged(caplog, expect_logged)
     for msg in fail_log_contains:
         assert msg in caplog.records[0].message
+
+
+def test_build_target_stage(rule_runner: RuleRunner) -> None:
```

Comment:
Is the next test a duplicate of this one?

Reply:
The setup is identical, but the execution and assert at the end is not. Expected `--target dev` vs `--target prod` due to `build_target_stage` option.

Reply:
Got it. Could you use `pytest.mark.parametrize`? Or factor it out with a helper function or variable?